addons/mail/static/src/web/messaging_menu/messaging_menu.js

```diff
@@ -117,39 +117,39 @@ export class MessagingMenu extends Component {
             if (a.correspondent === this.store.odoobot) {
                 return 1;
             }
-            if (!a.newestPersistentMessage?.datetime) {
+            if (a.hasNeedactionMessages && !b.hasNeedactionMessages) {
                 return -1;
             }
-            if (!b.newestPersistentMessage?.datetime) {
+            if (b.hasNeedactionMessages && !a.hasNeedactionMessages) {
+                return 1;
+            }
+            if (a.message_unread_counter > 0 && b.message_unread_counter === 0) {
+                return -1;
+            }
+            if (b.message_unread_counter > 0 && a.message_unread_counter === 0) {
+                return 1;
+            }
+            if (!a.newestPersistentMessage?.datetime) {
                 return 1;
             }
+            if (!b.newestPersistentMessage?.datetime) {
+                return -1;
+            }
             return b.newestPersistentMessage.datetime - a.newestPersistentMessage.datetime;
         });
         const previews = [];
         for (const thread of threads) {
             const { newestMessage, newestNeedactionMessage } = thread;
-            if (thread.is_pinned) {
-                const message = newestMessage;
+            if (thread.is_pinned || newestNeedactionMessage) {
                 previews.push({
                     id: `preview-${thread.localId}`,
```

Comment:
if there is a single type of preview now it looks like the `preview-` prefix is no longer necessary, and possible id could be removed in favor of just `thread.localId` in template?